### ISSUE TITLE
[Feature] Support import external proto in goctl.

### ISSUE THREAD

**Issue by chaozwn** [zeromicro/go-zero#4268]
**Is your feature request related to a problem? Please describe.**
In some scenarios, we may need to perform automatic validation of rpc parameters. for example, `protovalidate-go` library is introduced to automatically validate req.
1. This is the proto that I generated. It contains an external proto file and a new validation syntax.
![image](https://github.com/user-attachments/assets/e120481d-3cca-41f6-8845-9ed193f05535)
Next, I can verify req by directly generating the code automatically.
![image](https://github.com/user-attachments/assets/f274ade7-0ae5-4118-ab66-84bc2a8141cd)

**Describe the solution you'd like**
1. In fact, I have tested that it is now perfectly possible to use `buf-cli` instead of `protoc` to generate go files for grpc and proto.
The following is my solution case.

**buf.gen.yaml**
```yaml
version: v2

plugins:
  - remote: buf.build/grpc/go
    out: ../
  - remote: buf.build/protocolbuffers/go
    out: ../
  - remote: buf.build/bufbuild/validate-go
    out: ../
```

**buf.yaml**
```yaml
version: v2

name: buf.build/tutorials/lint
breaking:
  use:
    - FILE
lint:
  use:
    - DEFAULT
  except:
    - SERVICE_SUFFIX
    - FIELD_LOWER_SNAKE_CASE
    - PACKAGE_LOWER_SNAKE_CASE
    - PACKAGE_SAME_SWIFT_PREFIX
    - PACKAGE_VERSION_SUFFIX
    - ENUM_NO_ALLOW_ALIAS
    - BASIC
    - RPC_REQUEST_STANDARD_NAME
    - RPC_RESPONSE_STANDARD_NAME

deps:
  - buf.build/bufbuild/protovalidate
```
> update dep
```shell
buf dep update
```
> generate grpc & proto file
```shell
buf generate --path ./app/todo/cmd/rpc/pb/todo.proto --output ./app/todo/cmd/rpc/pb
```
![image](https://github.com/user-attachments/assets/2b0b4b6e-65e0-4308-84d4-20b684d77191)

In summary, we can get exactly the same grpc file and pb.go file with proto introduced.

But now there is a very fatal problem, I have import other proto files in proto early, resulting in the system's built-in generation function directly reported an error.

![image](https://github.com/user-attachments/assets/52f8cfc7-ba2e-45e4-91a3-72f82561aca2)

**Describe alternatives you've considered**

So I wonder if this syntax can be supported. `goctl rpc buf xx.proto --xxx`. If you are OK with my idea, I can try to submit a pr along these lines.

**Additional context**

**Comment by kesonan** [comment]
maybe you can try with -I $external_path, `goctl rpc protoc $proto ... -I $external_path --zrpc_out $dir --style=goZero --home $home -m`

**Comment by chaozwn** [comment]
> maybe you can try with -I $external_path, `goctl rpc protoc $proto ... -I $external_path --zrpc_out $dir --style=goZero --home $home -m`

first this directive has no way of taking effect, and the external proto code is not compiled by default

**Comment by fireice009** [comment]
I make it work. You also need `-I` for the path of .proto file. e.g. `goctl rpc protoc -I ../../common/protobuf/ -I api/proto api/proto/user.proto  --go_out=api --go-grpc_out=api --zrpc_out=api --client=true -m`

**Comment by fireice009** [comment]
I get a issue on [gateway](https://go-zero.dev/docs/tutorials/gateway/grpc) with importing `[validate.proto](https://github.com/bufbuild/protovalidate)`. 
1. Execute commands and got 3 *.pb files.
```shell
	protoc -I ../../common/protobuf/ -I api/proto --descriptor_set_out=api/asset/user.pb api/proto/user.proto
	protoc -I ../../common/protobuf/ --descriptor_set_out=api/asset/validate.pb ../../common/protobuf/validate.proto
	protoc -I ../../common/protobuf/google/protobuf --descriptor_set_out=api/asset/google/protobuf/descriptor.pb ../../common/protobuf/google/protobuf/descriptor.proto   # try 1
	protoc -I ../../common/protobuf/google/protobuf --descriptor_set_out=api/asset/descriptor.pb ../../common/protobuf/google/protobuf/descriptor.proto   # try 2

```
2. gateway config
```yaml
      ProtoSets:
        - asset/user.pb
        - asset/validate.pb
        - asset/google/protobuf/descriptor.pb  # try 1
        - asset/descriptor.pb # try 2
```

4. Run gateway service, and get the error msg  `no descriptor found for "google/protobuf/descriptor.proto"`

**Comment by fireice009** [comment]
> I get a issue on [gateway](https://go-zero.dev/docs/tutorials/gateway/grpc) with importing [validate.proto](https://github.com/bufbuild/protovalidate).
> 
> 1. Execute commands and got 3 *.pb files.
> 
> ```shell
> 	protoc -I ../../common/protobuf/ -I api/proto --descriptor_set_out=api/asset/user.pb api/proto/user.proto
> 	protoc -I ../../common/protobuf/ --descriptor_set_out=api/asset/validate.pb ../../common/protobuf/validate.proto
> 	protoc -I ../../common/protobuf/google/protobuf --descriptor_set_out=api/asset/google/protobuf/descriptor.pb ../../common/protobuf/google/protobuf/descriptor.proto   # try 1
> 	protoc -I ../../common/protobuf/google/protobuf --descriptor_set_out=api/asset/descriptor.pb ../../common/protobuf/google/protobuf/descriptor.proto   # try 2
> ```
> 
> 2. gateway config
> 
> ```yaml
>       ProtoSets:
>         - asset/user.pb
>         - asset/validate.pb
>         - asset/google/protobuf/descriptor.pb  # try 1
>         - asset/descriptor.pb # try 2
> ```
> 
> 3. Run gateway service, and get the error msg  `no descriptor found for "google/protobuf/descriptor.proto"`

forget it. it work by adding ` --include_imports `
